sshcode_test.go

```diff
@@ -0,0 +1,314 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"strconv"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.coder.com/retry"
+	"golang.org/x/crypto/ssh"
+)
+
+func TestSSHCode(t *testing.T) {
+	// Avoid opening a browser window.
+	err := os.Unsetenv("DISPLAY")
+	require.NoError(t, err)
+
+	sshPort, err := randomPort()
+	require.NoError(t, err)
+
+	// start up our jank ssh server
+	trassh(t, sshPort)
+
+	localPort := randomPortExclude(t, sshPort)
+	require.NotEmpty(t, localPort)
+
+	remotePort := randomPortExclude(t, sshPort, localPort)
+	require.NotEmpty(t, remotePort)
+
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		err := sshCode("127.0.0.1", "", options{
+			sshFlags:   testSSHArgs(sshPort),
+			localPort:  localPort,
+			remotePort: remotePort,
+		})
+		require.NoError(t, err)
+	}()
+
+	waitForSSHCode(t, localPort, time.Second*30)
+	waitForSSHCode(t, remotePort, time.Second*30)
+
+	out, err := exec.Command("pkill", "codessh").CombinedOutput()
```

Comment:
is this necessary.

Reply:
sshcode won't exit otherwise as it waits for a sigterm or the long-running ssh process to exit

Reply:
Ah. Wouldn't you want `os.Pid` here with `kill`? the go test binary can have a weird name.

Reply:
Nah this doesn't kill the `go test` process this kills the `code-server` process. Tests that sshcode exits gracefully when the `coder-server` process is terminated

Reply:
Why not `pkill code-server`? Why `codessh`?

Reply:
The process is called `codessh`, when we start it

Reply:
Honestly having it called `codessh` is pretty confusing, there's nothing different about it to warrant a different name

Reply:
I'm confused, I don't see a `codessh` anywhere.

Reply:
https://github.com/cdr/sshcode/blob/test/main.go#L133

Reply:
mm got it. def a good idea to rename, very confusing.